lib/miq_automation_engine/engine/miq_ae_engine.rb

```diff
@@ -27,71 +27,63 @@ def self.deliver_queue(args, options = {})
+  end
+
+  private_class_method def self.automate_attrs_from_options(options)
+    automate_attrs = options[:attrs].dup
+    automate_attrs['User::user']        = options[:user_id]           unless options[:user_id].nil?
+    automate_attrs[:ae_state]           = options[:state]             unless options[:state].nil?
+    automate_attrs[:ae_fsm_started]     = options[:ae_fsm_started]    unless options[:ae_fsm_started].nil?
+    automate_attrs[:ae_state_started]   = options[:ae_state_started]  unless options[:ae_state_started].nil?
+    automate_attrs[:ae_state_retries]   = options[:ae_state_retries]  unless options[:ae_state_retries].nil?
+    automate_attrs['ae_state_data']     = options[:ae_state_data]     unless options[:ae_state_data].nil?
+    automate_attrs['ae_state_previous'] = options[:ae_state_previous] unless options[:ae_state_previous].nil?
+    automate_attrs
+  end
+
+  private_class_method def self.create_automation_object_options(options, vmdb_object)
+    automation_object_options = {}
+    automation_object_options[:vmdb_object] = vmdb_object                unless vmdb_object.nil?
+    automation_object_options[:class]       = options[:class_name]       unless options[:class_name].nil?
+    automation_object_options[:namespace]   = options[:namespace]        unless options[:namespace].nil?
+    automation_object_options[:fqclass]     = options[:fqclass_name]     unless options[:fqclass_name].nil?
+    automation_object_options[:message]     = options[:automate_message] unless options[:automate_message].nil?
+    automation_object_options
+  end
+
+  private_class_method def self.change_options_by_ws(options, ws)
+    options[:state]             = ws.root['ae_state'] || options[:state]
+    options[:ae_fsm_started]    = ws.root['ae_fsm_started']
+    options[:ae_state_started]  = ws.root['ae_state_started']
+    options[:ae_state_retries]  = ws.root['ae_state_retries']
+    options[:ae_state_data]     = YAML.dump(ws.persist_state_hash) unless ws.persist_state_hash.empty?
+    options[:ae_state_previous] = YAML.dump(ws.current_state_info) unless ws.current_state_info.empty?
+  end
+
+  def self.deliver(*args)
+    options     = options_from_args(args)
+    user_obj    = ae_user_object(options)
+    state       = options[:state]
+    vmdb_object = nil
+    ae_result   = 'error'
 
     begin
-      object_name = "#{object_type}.#{object_id}"
+      object_name = "#{options[:object_type]}.#{options[:object_id]}"
       _log.info("Delivering #{options[:attrs].inspect} for object [#{object_name}] with state [#{state}] to Automate")
-      automate_attrs = options[:attrs].dup
+      automate_attrs = automate_attrs_from_options(options)
 
-      if object_type
-        vmdb_object = object_type.constantize.find_by(:id => object_id)
-        automate_attrs[create_automation_attribute_key(vmdb_object)] = object_id
+      if options[:object_type]
+        vmdb_object = options[:object_type].constantize.find_by(:id => options[:object_id])
```

Comment:
@pkomanek Part of this refactoring was to incorporate the change to use `find_by!` from https://github.com/ManageIQ/manageiq-automation_engine/pull/84/files on this line and make it easier to test.

Can you create a follow-up PR to introduce this change and tests?